cmd/tetragon/main.go

```diff
@@ -353,23 +354,46 @@ func startExporter(ctx context.Context, server *server.Server) error {
 		writer.FileMode = os.FileMode(0600)
 	}
 
-	if exportFileRotationInterval != 0 {
-		log.WithField("duration", exportFileRotationInterval).Info("Periodically rotating JSON export files")
+	finfo, err := os.Stat(filepath.Clean(exportFilename))
+	if err == nil && finfo.IsDir() {
+		// Error if exportFilename points to a directory
+		return fmt.Errorf("passed export JSON logs file point to a directory")
+	}
+	logFile := filepath.Base(exportFilename)
+	logsDir, err := filepath.Abs(filepath.Dir(filepath.Clean(exportFilename)))
+	if err != nil {
+		log.WithError(err).Warnf("Failed to get absolute path of exported JSON logs '%s'", exportFilename)
+		// Do not fail; we let lumberjack handle this. We want to
+		// log the rotate logs operation.
+		logsDir = filepath.Dir(exportFilename)
+	}
+	if exportFileRotationInterval > 0 {
```

Comment:
What happens if `exportFileRotationInterval == 0`? There seems to be a check initially but it was removed.

nit:
```
I'd also suggest we use, instead of two branches.

```
if (exportFileRotationInterval < 0) {
  return ...
}
...
```

Reply:
for exportFileRotationInterval == 0 I assume that also equals default value 0 which converts to 0seconds, and we should not rotate under this condition.

Ok will do that change now.

Reply:
Ok updated! changed the order same at the end... btw the logs should also be rotated by size --export-file-max-size-mb default is 10mb. Not related here and minor improvement I think since we can export logs to real storage, we should make it smarter taking GB annotation too...